pkg/cmd/roachtest/tests/backup.go

```diff
@@ -639,51 +645,60 @@ func registerBackup(r registry.Registry) {
 		},
 	})
 
-	r.Add(registry.TestSpec{
-		Name:              "backup/assume-role/AWS",
-		Owner:             registry.OwnerBulkIO,
-		Cluster:           r.MakeClusterSpec(3),
-		EncryptionSupport: registry.EncryptionMetamorphic,
-		Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
-			if c.Spec().Cloud != spec.AWS {
-				t.Skip("backup assumeRole is only configured to run on "+spec.AWS, "")
-			}
-
-			q := make(url.Values)
-			expect := map[string]string{
-				"AWS_ACCESS_KEY_ID_ASSUME_ROLE":     amazon.AWSAccessKeyParam,
-				"AWS_SECRET_ACCESS_KEY_ASSUME_ROLE": amazon.AWSSecretParam,
-				"AWS_ROLE_ARN":                      amazon.AWSRoleArnParam,
-				KMSRegionAEnvVar:                    amazon.KMSRegionParam,
-			}
-			for env, param := range expect {
-				v := os.Getenv(env)
-				if v == "" {
-					t.Errorf("env variable %s must be present to run the KMS test", env)
+	for _, item := range []struct {
+		cloudProvider string
+		machine       string
+	}{
+		{cloudProvider: "GCS", machine: spec.GCE},
+		{cloudProvider: "AWS", machine: spec.AWS},
+	} {
+		item := item
+		r.Add(registry.TestSpec{
+			Name:              fmt.Sprintf("backup/assume-role/%s", item.cloudProvider),
+			Owner:             registry.OwnerBulkIO,
+			Cluster:           r.MakeClusterSpec(3),
+			EncryptionSupport: registry.EncryptionMetamorphic,
+			Run: func(ctx context.Context, t test.Test, c cluster.Cluster) {
+				if c.Spec().Cloud != item.machine {
+					t.Skip("backup assumeRole is only configured to run on "+item.machine, "")
 				}
-				q.Add(param, v)
-			}
-			q.Add(cloudstorage.AuthParam, cloudstorage.AuthParamAssume)
 
-			rows := 100
-			dest := importBankData(ctx, rows, t, c)
+				rows := 100
 
-			m := c.NewMonitor(ctx)
-			m.Go(func(ctx context.Context) error {
-				t.Status(`running backup`)
-				kmsURI, err := getAWSKMSAssumeRoleURI(KMSRegionAEnvVar, KMSKeyARNAEnvVar)
-				if err != nil {
-					return err
+				dest := importBankData(ctx, rows, t, c)
+
+				var backupPath, kmsURI string
+				var err error
+				switch item.cloudProvider {
+				case "AWS":
+					if backupPath, err = getAWSBackupPath(dest); err != nil {
+						t.Fatal(err)
+					}
+					if kmsURI, err = getAWSKMSAssumeRoleURI(); err != nil {
+						t.Fatal(err)
+					}
+				case "GCS":
+					if backupPath, err = getGCSBackupPath(dest); err != nil {
+						t.Fatal(err)
+					}
+					if kmsURI, err = getGCSKMSAssumeRoleURI(); err != nil {
+						t.Fatal(err)
+					}
 				}
-				c.Run(ctx, c.Node(1), fmt.Sprintf(
-					"./cockroach sql --insecure -e \"BACKUP bank.bank TO 's3://cockroachdb-backup-testing/%s?%s' WITH KMS='%s'\"",
-					dest, q.Encode(), kmsURI))
-				return nil
-			})
-			m.Wait()
-		},
-	})
 
+				m := c.NewMonitor(ctx)
+				m.Go(func(ctx context.Context) error {
+					t.Status(`running backup`)
+					c.Run(ctx, c.Node(1), fmt.Sprintf(
+						"./cockroach sql --insecure -e \"BACKUP bank.bank TO '%s' WITH KMS='%s'\"",
```

Comment:
nit: can we use the `BACKUP INTO` syntax? We want to get rid of `TO` soonish so we'll have to cleanup all roachtests eventually.

Reply:
Is it useful to RESTORE from the backup and check if the table has expected rows as well? It'll be a more complete test imo, wdyt?